### DISCUSSION TITLE
Shopping List: Add Aisle field to items and also group and sort by aisle

### DISCUSSION THREAD

**Discussion by diedrichg** [mealie-recipes/mealie#2680] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

I would love to have the ability to add the aisle number (or name) to a list item! This is incredibly handy when at your grocery store or mega warehouse.

### Please provide a concise description of the feature that would resolve your issue.

- When adding or editing a shopping list item please add a field for Aisle Number but also have the ability to use a Aisle Name as well. This way someone can just reference the store's aisle number or if they prefer, the aisle name (e.g. Baking).
- Sorting by Aisle number or name will also be needed
- Grouping by Aisle number or name will also be needed
- Alphabetical sorting within a group would be helpful but someone may want to have a custom order (say - from front of aisle to back of aisle as they walk from one direction of the store or other)

### Please consider and list out some caveats or tradeoffs made in your design decision

Some people may not want to group by aisle number and may just want to sort by aisle only.
Some people may want a custom item order within the aisle group if they intend to shop the aisle from front to back or back to front.

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by michael-genson** [comment]
You can do this already with labels:
https://github.com/mealie-recipes/mealie/pull/2090

If you parse your recipes into foods, and your foods have labels, then adding them to the shopping list will let you organize by label. You can also manually add labels to one-off items in the editor

**Reply by diedrichg** [reply]
Sweet. I see that now - Labels are quite hidden. Very powerful system as I setup all the aisles and then created some foods with the labels. I then was successful in adding the foods and the labels were already there waiting for me to sort my list. Thanks for the help!

I'll submit a FR to add Labels to the main menu so it's easier to maintain.
